app/controllers/course_user_data_controller.rb

```diff
@@ -22,41 +22,42 @@ def new
   def create
     cud_parameters = cud_params
     @newCUD = @course.course_user_data.new(cud_parameters)
-    # do pre-validation of required fields
-    # must have email, and first OR last name
-    if cud_parameters[:user_attributes][:email].blank? ||
-       (cud_parameters[:user_attributes][:first_name].blank? &&
-       cud_parameters[:user_attributes][:last_name].blank?)
-      flash[:error] = "Error enrolling user: You must enter a valid email, and a first or last " \
-        "name to create a new student"
-      redirect_to(action: "new") && return
-    end
-    # check user existence
+
     email = cud_parameters[:user_attributes][:email]
     user = User.where(email: email).first
     if user.nil?
       # user is new
-      user = User.roster_create(email,
-                                cud_parameters[:user_attributes][:first_name],
-                                cud_parameters[:user_attributes][:last_name],
-                                "", "", "")
-      if user
-        @newCUD.user = user
+      # do pre-validation of required fields
+      # must have email, and first OR last name
+      # check user existence
```

Comment:
nit: this comment belongs to above the line `email = cud_parameters[:user_attributes][:email]`